debian/postinst

```diff
@@ -58,9 +58,7 @@ ln -s /opt/delphix/server/etc/nginx/nginx.conf ./etc/nginx/nginx.conf
 
 find . -print0 | cpio --null --create --format=newc | gzip -7 >"$target" 2>/dev/null
 
-mount -t zfs rpool/grub /mnt
-/usr/sbin/grub-mkconfig -o /mnt/boot/grub/grub.cfg
-umount /mnt
+/usr/sbin/grub-mkconfig
```

Comment:
How does this work given that we only mount grub onto `/mnt` so this would update a file that is not used by anything? Wouldn't we want to have the logic that we had before but make sure that we are not trying to run this in an upgrade container?

Reply:
> so this would update a file that is not used by anything? 

Correct.

> Wouldn't we want to have the logic that we had before but make sure that we are not trying to run this in an upgrade container?

We could, but IMO, having this package know about how we have a separate `rpool/grub` dataset on our Delphix appliance is a layering violation.

This same issue is present for any and all other packages that may attempt to update the grub configuration, for example when we install a new kernel package. In those cases, they also update what they think is the grub configuration, but is actually an unused file. This was by design, to try and prevent cases like this bug, where software running in the container may corrupt data outside of the container.

To account for these cases, we "manually" update the grub configuration after all packages are installed/upgraded (see [here](https://github.com/delphix/appliance-build/blob/master/upgrade/upgrade-scripts/upgrade#L177), [here](https://github.com/delphix/appliance-build/blob/master/upgrade/upgrade-scripts/upgrade#L288), [here](https://gitlab.delphix.com/app/dlpx-app-gate/blob/master/appliance/server/upg/src/main/java/com/delphix/appliance/server/upg/InPlaceUpgrade.java#L118), and [here](https://gitlab.delphix.com/app/dlpx-app-gate/blob/master/appliance/server/upg/src/main/java/com/delphix/appliance/server/upg/NotInPlaceUpgrade.java#L114)), and that will also handle the case of updating the grub configuration after we've upgraded the recovery-environment package.

This, by design, keeps the knowledge of the fact that we have a unique grub configuration/dataset out of individual packages, and places the responsibility to keep the grub configuration correct up to our Delphix specific upgrade logic.

Reply:
Further, I'm not sure we actually could rely on conditional logic like you're suggesting; for example, if we did that, how would that work for the case of not-in-place upgrades? In that case, we only install the package within a containerized environment, so we'd never actually update the grub configuration to point to the new root dataset, right?

Thus, I think keeping the design this way (i.e. not relying on packages to update the grub configuration) not only is better in terms of the layering of our appliance, but I think it actually results in a simpler solution in all of our various modes for upgrade and rollback.

Reply:
Given that this doesn't do anything, what's the value in keeping it in the postinst script?

Reply:
Okay, I'm fine with that, but I agree with George that in that case I don't see a value of keeping this line in the script.

Reply:
The theory that this change is based on is that we shouldn't be relying on the specific location we store our boot configuration in the postinst script for this package that is nominally not delphix specific. As a result, we simply use the default mkconfig; on the Delphix Engine, this file happens to get ignored, and we will regen our actually grub config at the end of the upgrade/appliance-build process.

Reply:
I see what you mean, that said the postinst script is already Delphix-specific, given that we are copying a bunch of Delphix-related files, such as:
```
sed -i 's@/opt/delphix/server/bin/supportlogin@/bin/sh@g' ./etc/passwd{,-}
mkdir -p ./opt/delphix/server/etc/{nginx.default,nginx}
ln -s /opt/delphix/server/etc/nginx/nginx.conf ./etc/nginx/nginx.conf
```

I think we'd want at the very least to add a comment explaining that this operation does nothing on a Delphix Engine, and that it relies on `grub-mkconfig` to be called later.

Reply:
> I see what you mean, that said the postinst script is already Delphix-specific

IMO, this is a good argument for removing the mkconfig command entirely. If the script is already tailored specifically for Delphix, then it likely doesn't make much sense trying to keep this single command for use cases outside of Delphix, since the script as a whole still will not work for non-DE systems.

Reply:
Yeah, I think whether or not we remove the command, a comment would be useful.

Reply:
I think leaving this in here just leads to confusion. That said, I'm trying to understand why we needed it here in the first place. Is there a gap where the grub config was not being updated correctly otherwise?